src/napari_threedee/data_models/paths.py

```diff
@@ -0,0 +1,123 @@
+import os
+from typing import List
+
+import napari
+import numpy as np
+import zarr
+from pydantic import BaseModel, validator
+
+from napari_threedee.annotators.base import N3dDataModel
+from napari_threedee.annotators.constants import N3D_METADATA_KEY, ANNOTATION_TYPE_KEY
+from napari_threedee.data_models.spline_sampler import SplineSampler
+from napari_threedee.annotators.paths.validation import (
+    validate_layer,
+    validate_n3d_zarr,
+)
+from napari_threedee.annotators.paths.constants import (
+    PATH_ID_FEATURES_KEY,
+    PATH_ANNOTATION_TYPE_KEY,
+    COLOR_CYCLE,
+)
+
+
+class N3dPath(BaseModel):
+    data: np.ndarray
+
+    class Config:
+        allow_mutation = False
+        arbitrary_types_allowed = True
+
+    @property
+    def ndim(self) -> int:
+        return self.data.shape[-1]
+
+    @property
+    def n_points(self) -> int:
+        return self.data.shape[0]
+
+    def sample(self, n: int = 10000) -> np.ndarray:
+        """Sample equidistant points between data points."""
+        sampler = SplineSampler(points=self.data)
+        return sampler(u=np.linspace(0, 1, num=n))
+
+    @validator('data', pre=True)
+    def ensure_float32_ndarray(cls, value):
+        return np.asarray(value, dtype=np.float32)
+
+    def __len__(self) -> int:
+        return len(self.data)
+
+
+class N3dPaths(N3dDataModel):
```

Comment:
Can we call this an `N3dPathCollection` or something that looks a little more different from `N3DPath`? I think it's a bit confusing as is.